plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer-v2.ts

```diff
@@ -510,23 +525,13 @@ export class SessionRecordingIngesterV2 {
         await this.batchConsumer?.stop()
 
         // Simulate a revoke command to try and flush all sessions
-        // The rebalance event should have done this but we do it again as an extra precaution and to await the flushes
-        await this.onRevokePartitions(
-            Object.keys(this.partitionAssignments).map((partition) => ({
-                partition: parseInt(partition),
-                topic: this.topic,
-            })) as TopicPartition[]
-        )
+        // There is a race between the revoke callback and this function - Either way one of them gets there and covers the revocations
+        void this.scheduleWork(this.onRevokePartitions(this.assignedTopicPartitions))
```

Comment:
you could `scheduleWork` against labels in a `Record<string, Promise[]>` and then this could wait on any existing revoke before triggering another for safety ... 

But that might be unnecessary so 🤷

Reply:
I thought about that as well but favoured this for simplicity sake. The async part of the revoke code is isolated now so it shouldn't be affected by races.... he says...